main.go

```diff
@@ -163,67 +168,99 @@ func delete(cmd *cobra.Command, args []string) error {
 }
 
 func listDbs(cmd *cobra.Command, args []string) error {
-	cc, err := client.NewClientWithDefaults()
-	if err != nil {
-		return err
+	dbs, err := getDbs()
+	for _, db := range dbs {
+		fmt.Println(db)
 	}
-	dd, err := cc.DataPath()
+	return err
+}
+
+// getDbs: returns a formatted list of available Skate DBs
+func getDbs() ([]string, error) {
+	filepath, err := getFilePath()
 	if err != nil {
-		return err
+		return nil, err
 	}
-	dbs, err := os.ReadDir(filepath.Join(dd, "kv"))
+	entries, err := os.ReadDir(filepath)
 	if err != nil {
-		return err
+		return nil, err
 	}
-	for _, d := range dbs {
-		if d.IsDir() {
-			fmt.Println("@" + d.Name())
+	var out []string
+	for _, e := range entries {
+		if e.IsDir() {
+			out = append(out, "@"+e.Name())
 		}
 	}
-	return nil
+	return out, nil
 }
 
-func deleteDb(cmd *cobra.Command, args []string) error {
-	n, err := nameFromArgs(args)
-	if err != nil {
-		return err
-	}
+// getFilePath: get the file path to the skate databases.
+func getFilePath(args ...string) (string, error) {
 	cc, err := client.NewClientWithDefaults()
 	if err != nil {
-		return err
+		return "", err
 	}
 	dd, err := cc.DataPath()
+	if err != nil {
+		return "", err
+	}
+	return filepath.Join(append([]string{dd, "kv"}, args...)...), err
+}
+
+// deleteDb: delete a Skate database.
+func deleteDb(cmd *cobra.Command, args []string) error {
+	// get client info
+	n, err := nameFromArgs(args)
 	if err != nil {
 		return err
 	}
-	dbs, err := os.ReadDir(filepath.Join(dd, "kv"))
+	path, err := getFilePath(n)
 	if err != nil {
 		return err
 	}
-
-	var found bool
-	for _, d := range dbs {
-		if d.IsDir() && d.Name() == n {
-			found = true
-			var confirmation string
-			fmt.Println(warningStyle.Render("are you sure you want to delete " + d.Name() + " and all its contents? (y/n)"))
-			fmt.Scanln(&confirmation)
-			if confirmation == "y" {
-				err := os.RemoveAll(filepath.Join(dd, "kv", d.Name()))
-				if err != nil {
-					return err
-				}
-			} else {
-				fmt.Println("did not delete " + d.Name())
-			}
+	_, err = os.Stat(path)
+	if n == "" || os.IsNotExist(err) {
+		fmt.Println(notFound(args[0]))
+	} else {
+		var confirmation string
+		fmt.Printf("are you sure you want to delete '%s' and all its contents?(y/n) ", warningStyle.Render(path))
+		fmt.Scanln(&confirmation)
+		if confirmation == "y" {
+			return os.RemoveAll(path)
 		}
-	}
-	if !found {
-		fmt.Println(args[0] + " does not exist")
+		fmt.Println("did not delete " + n)
 	}
 	return nil
 }
 
+// notFound: displays a not found message with suggestions.
+func notFound(v string) string {
```

Comment:
I might still refactor this further to account for the errors that I've ignored here (heh) and make this return the descriptive error type ayman mentioned